src/agent/src/cdh.rs

```diff
@@ -49,13 +49,101 @@ impl CDHClient {
     }
 
     pub async fn unseal_env(&self, env: &str) -> Result<String> {
-        let (key, value) = env.split_once('=').unwrap();
+        let (key, value) = env.split_once('=').unwrap_or(("", ""));
```

Comment:
I think it's safer to catch the error, log a warning message and return. Otherwise it will silently change the envvars, can cause bugs hard to find.

Reply:
If the secret isn't of the form `key=value`, then the next conditional where we check for the secret prefix will fail and the secret won't be modified. The only case we should be modifying the env is if it is a sealed secret. I'm not sure if it's more clear to return immediately in that case or fall through to the end of the function.

Reply:
oh, never mind, for some reason I thought that `*env.to_owned()).to_string()` was actually returning (the modified) `key=value`.

Return immediately will save one conditional.Maybe the compiler will optimize that....

Reply:
I changed this to be an `if let`, which seems a lot clearer.